packages/quantic/force-app/main/default/lwc/quanticSearchBoxSuggestionsList/quanticSearchBoxSuggestionsList.js

```diff
@@ -41,87 +93,200 @@ export default class QuanticSearchBoxSuggestionsList extends LightningElement {
   @api
   selectionDown() {
     this.selectionIndex++;
-    if (this.selectionIndex >= this.suggestions.length) {
+    if (this.selectionIndex >= this.allOptions.length) {
       this.selectionIndex = 0;
     }
+    return this.allOptionsHTMLElements[this.selectionIndex].getAttribute('id');
   }
 
   /**
    * Return the currently selected suggestion.
-   * @returns {Suggestion}
+   * @returns {Object}
    */
   @api
   getCurrentSelectedValue() {
-    if (this.selectionIndex > -1) {
-      return this.suggestions[this.selectionIndex];
+    if (this.allOptions?.[this.selectionIndex]) {
+      return {
+        value: this.allOptions[this.selectionIndex].rawValue,
+        isClearRecentQueryButton:
+          this.allOptions[this.selectionIndex].isClearRecentQueryButton,
+        isRecentQuery: this.allOptions[this.selectionIndex].isRecentQuery,
+      };
     }
-    return undefined;
-  }
-
-  /**
-   * Reset the current selection.
-   */
-  @api
-  resetSelection() {
-    this.selectionIndex = -1;
+    return null;
   }
 
   /** @type {number} */
-  @track selectionIndex = -1;
+  selectionIndex = -1;
   /** @type {boolean} */
   initialRender = true;
+  /** @type {string} */
+  previousQuery = '';
 
   renderedCallback() {
-    if (this.selectionIndex > -1) {
-      this.emitSuggestionHighlighted();
-    }
     if (this.initialRender) {
-      const listboxId = this.template.querySelector('div').getAttribute('id');
-      const suggestionListEvent = new CustomEvent('suggestionlistrender', {
-        detail: listboxId,
-        bubbles: true,
-        composed: true,
-      });
-      this.dispatchEvent(suggestionListEvent);
+      this.suggestionsAriaLiveMessage = AriaLiveRegion(
+        'suggestions',
+        this,
+        true
+      );
+      this.sendSuggestionListIdToInput();
```

Comment:
Actually this component renders multiple times, like if there are suggestions we display it otherwise it's completely removed from the DOM.
And yes that's the only way we have so far to deal with this.